src/clients/BundleDataClient/BundleDataClient.ts

```diff
@@ -831,24 +836,13 @@ export class BundleDataClient {
             return;
           }
 
-          // If deposit block is within origin chain bundle block range, then save as bundle deposit.
-          // If deposit is in bundle and it has expired, additionally save it as an expired deposit.
-          // Note: if the `depositor` field in the expired deposit is an invalid address, e.g. a bytes32 address on an EVM
-          // chain, then the deposit refund is invalid and ignored.
-          // If deposit is not in the bundle block range, then save it as an older deposit that
-          // may have expired.
-          if (deposit.blockNumber >= originChainBlockRange[0] && deposit.blockNumber <= originChainBlockRange[1]) {
-            // Deposit is a V3 deposit in this origin chain's bundle block range and is not a duplicate.
+          // Evaluate all expired deposits after fetching fill statuses,
+          // since we can't know for certain whether an expired deposit was filled a long time ago.
+          if (deposit.blockNumber >= originChainBlockRange[0]) {
+            bundleDepositHashes.push(relayDataHash);
             updateBundleDepositsV3(bundleDepositsV3, deposit);
-            // We don't check that fillDeadline >= bundleBlockTimestamps[destinationChainId][0] because
-            // that would eliminate any deposits in this bundle with a very low fillDeadline like equal to 0
-            // for example. Those should be impossible to create but technically should be included in this
-            // bundle of refunded deposits.
-            if (deposit.fillDeadline < bundleBlockTimestamps[destinationChainId][1]) {
-              expiredBundleDepositHashes.add(relayDataHash);
-            }
           } else if (deposit.blockNumber < originChainBlockRange[0]) {
-            olderDepositHashes.add(relayDataHash);
+            olderDepositHashes.push(relayDataHash);
           }
         });
       }
```

Comment:
Line 880 needs to be 
```
.filter((fill) => fill.blockNumber <= destinationChainBlockRange[1] && !isZeroValueFillOrSlowFillRequest(fill)),
```
I believe.

Reply:
yep this got reverted due to a merge conflict, i've added back